### PR TITLE
Messages collation

### PR DESCRIPTION
**Pull request by kwinto**
This PR implements message collation: subsequent messages from the same source are rendered without header (if they are within a 1-minute limit from the previous message).

See #55736 in Azure for ref.

Also moved GitHub action configs to a proper directory. This should enable test pipeline in the repo after the PR is merged.

Changes necessary to test this in Webchat are already on the v3 branch.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -29,6 +29,17 @@ export const getWebchatButtonLabel: getWebchatButtonLabel = button => {
 	return title;
 };
 
+export const isMessageCollatable = (message: WebchatMessage, prevMessage?: WebchatMessage) => {
```
Comment (fabclj) [inline src/utils.ts:32]: Please use IMessage interface here

File: src/utils.ts [hunk 2]
```diff
@@ -29,6 +29,17 @@ export const getWebchatButtonLabel: getWebchatButtonLabel = button => {
...
+	const COLLATION_LIMIT = 1000 * 60; // 60 sec
+
+	const difference = Number(message?.timestamp) - Number(prevMessage?.timestamp);
+	return (
+		isNaN(difference) === false &&
+		difference < COLLATION_LIMIT &&
+		prevMessage.source === message.source
```
Comment (fabclj) [inline src/utils.ts:39]: pelase add safe null check here

File: test/Collation.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { render } from "@testing-library/react";
+import { it, describe, expect } from "vitest";
+import Message from "../src/messages/Message";
+
+describe("Collation", () => {
+	it("collates if timestamp is in limit", () => {
+		const message1 = { text: "Hello", source: "bot", timestamp: Date.now() - 1000 * 50 };
```
Comment (fabclj) [inline test/Collation.spec.tsx:7]: here you can simple infer IMessage like in the other specs:
`const message1 = { text: "Hello", source: "bot", timestamp: Date.now() - 1000 * 50 } as unknown as IMessage;`

File: src/demo.tsx [hunk 4]
```diff
@@ -112,19 +133,30 @@ const messages: MessageProps[] = [
...
 	{
 		message: {
-			source: "bot",
-			text: "This messaged is with previous with disableHeader prop",
-			avatarName: "Cognigy",
-		},
-		disableHeader: true,
-	},
-	{
-		message: image as IMessage,
```
Comment (fabclj) [inline src/demo.tsx:122]: why you removed the fixture here? is much less chaotic using directly the fixture when possible

File: src/messages/Message.tsx [hunk 5]
```diff
@@ -16,16 +17,20 @@ export interface MessageProps {
 	config?: IWebchatConfig;
 	disableHeader?: boolean;
 	message: IMessage;
-	plugins?: MatchConfig[];
 	onEmitAnalytics?: (event: string, payload?: unknown) => void;
+	plugins?: MatchConfig[];
+	prevMessage?: WebchatMessage;
```
Comment (fabclj) [inline src/messages/Message.tsx:22]: here you have to use IMessage interface

File: src/utils.ts (hunk 1 above)
Comment (kwinto) [reply src/utils.ts]: fixed

File: test/Collation.spec.tsx (hunk 3 above)
Comment (kwinto) [reply test/Collation.spec.tsx]: fixed

File: src/demo.tsx (hunk 4 above)
Comment (kwinto) [reply src/demo.tsx:122]: mistook by conflict resolution, fixed

File: src/messages/Message.tsx (hunk 5 above)
Comment (kwinto) [reply src/messages/Message.tsx]: fixed

File: src/utils.ts (hunk 2 above)
Comment (kwinto) [reply src/utils.ts]: fixed

**Comment by fabclj** [comment]
@kwinto I did functional test, and looks working to me.
Just a question: the collation is working for all message types. Is it expected? Or maybe the idea was just to apply collation only to text messages?

**Comment by kwinto** [comment]
> @kwinto I did functional test, and looks working to me. Just a question: the collation is working for all message types. Is it expected? Or maybe the idea was just to apply collation only to text messages?

Yes, it should be independent of message type.

**Review by fabclj — APPROVED** [review]

**Pull request merged**